include/mppic/utils.hpp

```diff
@@ -125,6 +129,65 @@ auto shortest_angular_distance(
   return normalize_angles(to - from);
 }
 
+/**
+ * @brief Evaluate furthest point idx of data.path which is
+ * nearset to some trajectory in data.trajectories
+ */
+inline size_t findPathFurthestReachedPoint(const models::CriticFunctionData & data)
+{
+  auto path_points = xt::view(data.path, xt::all(), xt::range(0, 2));
+
+  auto last_points_ext =
+    xt::view(data.trajectories, xt::all(), -1, xt::newaxis(), xt::range(0, 2));
+  auto distances = xt::norm_l2(last_points_ext - path_points, {2});
+  size_t max_id_by_trajectories = 0;
+  double min_distance_by_path = std::numeric_limits<double>::max();
+
+  for (size_t i = 0; i < distances.shape(0); i++) {
+    size_t min_id_by_path = 0;
+    for (size_t j = 0; j < distances.shape(1); j++) {
+      if (min_distance_by_path < distances(i, j)) {
+        min_distance_by_path = distances(i, j);
+        min_id_by_path = j;
+      }
+    }
+    max_id_by_trajectories = std::max(max_id_by_trajectories, min_id_by_path);
+  }
+  return max_id_by_trajectories;
+}
+
+
+inline void setPathFurthestPointIfNotSet(models::CriticFunctionData & data)
+{
+  if (!data.furthest_reached_path_point) {
+    data.furthest_reached_path_point = findPathFurthestReachedPoint(data);
+  }
+}
+
+inline auto pose_point_angle(geometry_msgs::msg::Pose pose, double point_x, double point_y)
```

Comment:
`posePointAngle` for styling to be the same

Reply:
done